src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   */
+  redecomp::RedecompMesh& GetRedecompMesh()
+  {
+    return GetUpdateData().redecomp_;
+  }
+
+  /**
+   * @brief Get the finite element space on the (coarse) boundary submesh
+   * 
+   * @return const mfem::ParFiniteElementSpace& 
+   */
+  const mfem::ParFiniteElementSpace& GetSubmeshFESpace() const
+  {
+    return *submesh_xfer_gridfn_.ParFESpace();
+  }
+
+  /**
+   * @brief Get the finite element space on the refined boundary submesh
+   * 
+   * @return const mfem::ParFiniteElementSpace* 
+   *
+   * @note nullptr if no refined mesh exists
+   */
+  const mfem::ParFiniteElementSpace* GetLORSubmeshFESpace() const
+  {
+    return lor_xfer_ ? lor_xfer_->GetLORGridFn().ParFESpace() : nullptr;
+  }
+
+  /**
+   * @brief Set the number of element subdivisions per dimension on the refined
+   * mesh
+   *
+   * @param lor_factor Number of element subdivisions per dimension
+   */
+  void SetLowOrderRefinedFactor(integer lor_factor);
+private:
+  /**
+   * @brief Creates and stores data that changes when the redecomp mesh is
+   * updated
+   */
+  struct UpdateData
+  {
+    /**
+     * @brief Construct a new UpdateData object
+     * 
+     * @param submesh Coarse boundary submesh of contact elements
+     * @param lor_submesh (Optional) Refined boundary submesh of contact elements
+     * @param parent_fes Vector finite element space on the original parent mesh
+     * @param submesh_gridfn Grid function on the (coarse) submesh used to
+     * temporarily store variables being transferred
+     * @param lor_xfer (Optional) Low-order grid function, refined mesh transfer
+     * object
+     * @param attributes_1 Set of boundary attributes for the first mesh
+     * @param attributes_2 Set of boundary attributes for the second mesh
+     * @param num_verts_per_elem Number of vertices on each element
+     */
+    UpdateData(
+      mfem::ParSubMesh& submesh,
+      mfem::ParMesh* lor_submesh,
+      const mfem::ParFiniteElementSpace& parent_fes,
+      mfem::ParGridFunction& submesh_gridfn,
+      SubmeshLORTransfer* lor_xfer,
+      const std::set<integer>& attributes_1,
+      const std::set<integer>& attributes_2,
+      integer num_verts_per_elem
+    );
+
+    /**
+     * @brief Redecomposed (refined) boundary element mesh
+     */
+    redecomp::RedecompMesh redecomp_;
+
+    /**
+     * @brief Parent to redecomp field transfer object
+     */
+    ParentRedecompTransfer vector_xfer_;
+
+    /**
+     * @brief Redecomp mesh element connectivity for the first mesh
+     */
+    axom::Array<integer, 2> conn_1_;
+
+    /**
+     * @brief Redecomp mesh element connectivity for the second mesh
+     */
+    axom::Array<integer, 2> conn_2_;
+
+    /**
+     * @brief Map from mesh 1 element indicies to redecomp mesh element indices
+     */
+    std::vector<integer> elem_map_1_;
+
+    /**
+     * @brief Map from mesh 2 element indicies to redecomp mesh element indices
+     */
+    std::vector<integer> elem_map_2_;
```

Comment:
See comment above for clarity suggestions and questions. Here, mainly is "mesh 2" an actual separate mesh or is it really "Map from contact surface 2 element indices to redecomp mesh element indices" ?

Indices (sp).

Reply:
Addressed above; comment fixed.